e2e/cypress/tests/integration/performance/channel_switch_spec.js

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Stage: @prod
+// Group: @channel
+
+import {measurePerformance} from './utils.js';
+
+describe('Channel switch performance test', () => {
+    let testUser;
+
+    before(() => {
+        cy.apiInitSetup().then(({team, user}) => {
+            testUser = user;
+
+            // # Login as test user and go to town square
+            cy.apiLogin(testUser);
+            cy.visit(`/${team.name}/channels/town-square`);
+        });
+    });
+
+    it('measures switching between two channels from LHS', async () => {
+        // # Invoke window object
+        await measurePerformance('channelLoad', 500, () => {
+            // # Switch channel to Off-topic
+
+            cy.get('#sidebarItem_off-topic').click({force: true});
+
+            // * Expect that the user is now in Off-Topic
+            expectActiveChannelToBe('Off-Topic', '/off-topic');
+        });
+    });
+});
+
+const expectActiveChannelToBe = (title, url) => {
+    // * Expect channel title to match title passed in argument
+    cy.get('#channelHeaderTitle').
+        should('be.visible').
+        and('contain.text', title);
```

Comment:
For channel switch, are we happy with just the channel header changing, or should we make sure the posts are visible? What is the input that makes the user believe the channel has finished loading? We should be looking for that.

Reply:
I will adjust this to also track for page load, CI performance has that built in. Thanks!

Reply:
i would suggest to be more accurate representation to this:
- create post and send the the above channel by another user before going to that channel
- once the switch is made to check if the posts are loaded and equal to what we sent before
- and the text box is in editable/writable state

Reply:
I think thats a good point but for follow up testing. Like in a separate 'it' block. I can add those